src/antsibull_core/utils/io.py

```diff
@@ -71,10 +72,17 @@ async def copy_file(
     flog.debug("Leave")
 
 
-async def write_file(filename: StrOrBytesPath, content: str) -> None:
+async def write_file(
+    filename: StrOrBytesPath, content: str, /, sanitize: bool = False
+) -> None:
+    """Dumps a generated text file to disk, allows sanitization on trimming spaces."""
     flog = mlog.fields(func="write_file")
     flog.debug("Enter")
 
+    if sanitize:
+        content = re.sub(r"[ \t]+$", "", content, flags=re.MULTILINE)
+        content = content.rstrip("\n\r") + "\n"
```

Comment:
I agree with Felix that perhaps this isn't the best place to do this, but maybe a separate `sanitize_trailing_spaces()` function in antsibull_core would make sense